core-api/src/main/java/com/optimizely/ab/config/parser/AudienceGsonDeserializer.java

```diff
@@ -67,19 +67,23 @@ private Condition parseConditions(List<Object> rawObjectList) {
                 List<Object> objectList = (List<Object>)rawObjectList.get(i);
                 conditions.add(parseConditions(objectList));
             } else {
-                LinkedTreeMap<String, String> conditionMap = (LinkedTreeMap<String, String>)rawObjectList.get(i);
-                conditions.add(new UserAttribute(conditionMap.get("name"), conditionMap.get("type"),
-                               conditionMap.get("value")));
+                LinkedTreeMap<String, ?> conditionMap = (LinkedTreeMap<String, ?>)rawObjectList.get(i);
+                conditions.add(new UserAttribute((String)conditionMap.get("name"), (String)conditionMap.get("type"),
+                        (String)conditionMap.get("match"), conditionMap.get("value")));
             }
         }
 
         Condition condition;
-        if (operand.equals("and")) {
-            condition = new AndCondition(conditions);
-        } else if (operand.equals("or")) {
-            condition = new OrCondition(conditions);
-        } else {
-            condition = new NotCondition(conditions.get(0));
+        switch (operand) {
+            case "and":
+                condition = new AndCondition(conditions);
+                break;
+            case "or":
+                condition = new OrCondition(conditions);
+                break;
+            default:
+                condition = new NotCondition(conditions.get(0));
```

Comment:
After discussing with @mjc1283: we should also return `null` instead of raising `IndexOutOfBoundsException` if `conditions.length == 0`.

Reply:
Is there ever going to be a audience without a condition (not to be confused with audience id list)?

Reply:
> Is there ever going to be a audience without a condition (not to be confused with audience id list)?

As we started to discuss in Slack: if an audience object has null or missing `conditions`, I think that's rather problematic and it's appropriate for the SDK to reject the datafile. On the other hand, if the `conditions` are empty (`[]`, `{}`, etc.) then it's fine to apply our recursive evaluation algorithm to that value.